src/Core/Product/Search/SortOrder.php

```diff
@@ -120,6 +120,9 @@ public function toString()
      */
     public static function newFromString($sortOrderConfiguration)
     {
+        if (count(explode('.', $sortOrderConfiguration)) !== 3) {
+            return new static('', '', 'asc');
+        }
```

Comment:
```php
 public static function newFromString($sortOrderConfiguration)
    {
       $sortParams =  explode('.', $sortOrderConfiguration);

       if (count($sortParams) !== 3) {
            return static::random();
        }

        list($entity, $field, $direction) = $sortParams;

        return new static($entity, $field, $direction);
    }
```

Reply:
I prefer the first solution. It's more readable.

Reply:
@PululuK why returning random ? Is it the default sort order ?

Reply:
@PululuK @Progi1984 I've commited your suggestion. Please add hacktoberfest label.

Reply:
The hacktoberfest label doesn't need to be added, we already tagged our repositories :wink:

Reply:
@PierreRambaud, it is what has been asked in the Build post [here](https://build.prestashop.com/news/hacktoberfest-2020-is-coming). Goal is to get an estimated number of the pull requests submitted in the context of this challenge, there is no way to have KPI on it today, cf. https://github.com/PrestaShop/prestashop-retro/issues/80.